pandas/core/arrays/arrow/array.py

```diff
@@ -427,6 +428,8 @@ def _box_pa_scalar(cls, value, pa_type: pa.DataType | None = None) -> pa.Scalar:
             pa_scalar = value
         elif isna(value):
             pa_scalar = pa.scalar(None, type=pa_type)
+        elif pa_type == pa.date32() and isinstance(value, datetime.datetime):
```

Comment:
@PedroVerardo this is a nice find, but if pyarrow supports it I don't think we should do anything special in pandas to handle this. Looks like we should just open the issue upstream